src/client/views/UpdatePattern.tsx

```diff
@@ -1,11 +1,76 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import noteService from "../services/pattern";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
+import { useForm } from "../utils/use-form";
+import patternService from "../services/pattern";
+import { IPattern } from "../utils/types";
 
 interface UpdatePatternProps {}
 
 const UpdatePattern = (props: UpdatePatternProps) => {
+	const [pattern, setPattern] = useState<IPattern | undefined>();
+	const [content, setContent] = useState<string>();
+	const [title, setTitle] = useState<string>();
+	const { id } = useParams();
+	const { state } = useLocation();
+	const navigate = useNavigate();
+
+	const handleChanges = (
+		e: React.ChangeEvent<HTMLInputElement & HTMLTextAreaElement>
+	) => {
+		setPattern((prev) => ({
+			...prev,
+			[e.target.name]: e.target.value,
+		}));
+	};
+
+	useEffect(() => {
+		if (!state) {
+			if (!id) return;
+			patternService.getOnePattern(id).then((pattern: IPattern) => {
+				setPattern(pattern);
+				setContent(pattern.content);
+			});
+			// .catch(e => Toast.error(e.message));
+		}
+	}, [id]);
+
+	const handleUpdate = (e: React.MouseEvent<HTMLButtonElement>) => {
+		e.preventDefault();
+		if (!id || !pattern) return;
+		const patternDTO = {
+			content: pattern.content,
+			author_id: pattern.author_id,
+			id,
+		};
+		noteService
+			.updatePattern(id, patternDTO)
+			.then(() => navigate(`/patterns/${id}`));
```

Comment:
```suggestion
		noteService.updatePattern(id, patternDTO)
			.then(() => navigate(`/patterns/${id}`));
```
This is a super minor nitpick, but usually you only need to write all the function calls on different lines if the first line is too long. If you like it better your way though you can keep it!

Reply:
I like the two lines because it's easier for me to read and my auto formatter does it that way anyway